validator/sawtooth_validator/journal/validation_rule_enforcer.py

```diff
@@ -62,118 +69,121 @@ def validate(self, block, state_root):
             rule_type = rule_type.strip()
             # NofX: Only N of transaction type X may be included in a block.
             if rule_type == "NofX":
-                valid = self._do_nofx(transactions, arguments)
+                valid = _do_nofx(transactions, arguments)
 
             # XatY: A transaction of type X must be in the block at position Y.
             elif rule_type == "XatY":
-                valid = self._do_xaty(transactions, arguments)
+                valid = _do_xaty(transactions, arguments)
 
             # local: A transaction must be signed by the same key as the block.
             elif rule_type == "local":
-                valid = self._do_local(
+                valid = _do_local(
                     transactions, expected_signer, arguments)
 
             if not valid:
                 return False
 
         return valid
 
-    def _do_nofx(self, transactions, arguments):
-        """
-        Only N of transaction type X may be included in a block. The first
-        argument must be interpretable as an integer. The second argument is
-        interpreted as the name of a transaction family. For example, the
-        string "NofX:2,intkey" means only allow 2 intkey transactions per
-        block.
-        """
-        try:
-            num, family = arguments.split(',')
-            limit = int(num.strip())
-        except ValueError:
-            LOGGER.warning("Ignore, NofX requires arguments in the format "
-                           "int,family not %s", arguments)
-            return True
-        count = 0
-        family = family.strip()
-        for txn in transactions:
-            header = TransactionHeader()
-            header.ParseFromString(txn.header)
-            if header.family_name == family:
-                count += 1
-
-            if count > limit:
-                LOGGER.debug("Too many transactions of type %s", family)
-                return False
 
+def _do_nofx(transactions, arguments):
```

Comment:
These low-level functions should pass error messages upwards rather than doing the logging themselves.